test/agentchat/contrib/test_gpt_assistant.py

```diff
@@ -62,5 +63,114 @@ def test_gpt_assistant_chat():
     assert len(analyst._openai_threads) == 0
 
 
+@pytest.mark.skipif(
+    sys.platform in ["darwin", "win32"] or skip_test,
+    reason="do not run on MacOS or windows or dependency is not installed",
+)
```

Comment:
If a test needs to call the model, you want to try to import openai in the beginning to check if it is available. Checkout `test_compressible_agent.py`

Reply:
Probably not needed since openai is imported in gptassistant.py, and will be checked by `skip_test`